phpunit/functional/Glpi/Inventory/InventoryTest.php

```diff
@@ -6534,7 +6535,7 @@ public function testDefaultStatesOnAddWithGlobalLock()
         $this->assertSame($other_states_id, $computer->fields['states_id']);
     }
 
-
+    #[RunInSeparateProcess] // TODO: fix this test, it shouldn't need an individual process
```

Comment:
I am not able to make it work. I tried to apply a global patch to reinit the rules list whenever a rule is created/updated. It works for other tests, but not for this one. Maybe it is not related to the rules.

Here is the patch I tried.
```diff
diff --git a/phpunit/functional/Glpi/Inventory/InventoryTest.php b/phpunit/functional/Glpi/Inventory/InventoryTest.php
index e8b12186c9..be5aa2e3d4 100644
--- a/phpunit/functional/Glpi/Inventory/InventoryTest.php
+++ b/phpunit/functional/Glpi/Inventory/InventoryTest.php
@@ -8316,8 +8316,6 @@ Compiled Tue 28-Sep-10 13:44 by prod_rel_team",
         <QUERY>INVENTORY</QUERY>
         </REQUEST>";
 
-        \SingletonRuleList::getInstance("RuleAsset", 0)->load = 0;
-        \SingletonRuleList::getInstance("RuleAsset", 0)->list = [];
         $this->doInventory($xml_source, true);
 
         //check created agent
diff --git a/src/Rule.php b/src/Rule.php
index 932e889964..3a1b78c072 100644
--- a/src/Rule.php
+++ b/src/Rule.php
@@ -1989,12 +1989,16 @@ JS
     {
         parent::post_addItem();
         $this->handleRankChange(true);
+
+        SingletonRuleList::resetInstances();
     }
 
     public function post_updateItem($history = true)
     {
         parent::post_updateItem($history);
         $this->handleRankChange();
+
+        SingletonRuleList::resetInstances();
     }
 
     /**
diff --git a/src/RuleAction.php b/src/RuleAction.php
index 5d4e91945e..5ef2f1c51a 100644
--- a/src/RuleAction.php
+++ b/src/RuleAction.php
@@ -119,6 +119,15 @@ class RuleAction extends CommonDBChild
                 'date_mod' => $_SESSION['glpi_currenttime']
             ]);
         }
+
+        SingletonRuleList::resetInstances();
+    }
+
+    public function post_updateItem($history = true)
+    {
+        parent::post_updateItem($history);
+
+        SingletonRuleList::resetInstances();
     }
 
     public function post_purgeItem()
diff --git a/src/RuleCriteria.php b/src/RuleCriteria.php
index 7730eff89c..679b7338be 100644
--- a/src/RuleCriteria.php
+++ b/src/RuleCriteria.php
@@ -116,6 +116,15 @@ class RuleCriteria extends CommonDBChild
                 'date_mod' => $_SESSION['glpi_currenttime']
             ]);
         }
+
+        SingletonRuleList::resetInstances();
+    }
+
+    public function post_updateItem($history = true)
+    {
+        parent::post_updateItem($history);
+
+        SingletonRuleList::resetInstances();
     }
 
     public function post_purgeItem()
diff --git a/src/SingletonRuleList.php b/src/SingletonRuleList.php
index 1e5c8366b7..90697ceb98 100644
--- a/src/SingletonRuleList.php
+++ b/src/SingletonRuleList.php
@@ -35,6 +35,11 @@
 
 class SingletonRuleList
 {
+    /**
+     * @var SingletonRuleList[][]
+     */
+    private static $instances = [];
+
     /** @var Rule[] */
     public $list = [];
    /// Items loaded ?
@@ -49,19 +54,24 @@ class SingletonRuleList
      *
      * @return SingletonRuleList unique instance of an object
      **/
-    public static function &getInstance($type, $entity)
+    public static function getInstance($type, $entity)
     {
-        //FIXME: can be removed when using phpunit 10 and process-isolation
-        if (defined('TU_USER')) {
-            $o = new self();
-            return $o;
+        if (!isset(self::$instances[$type][$entity])) {
+            self::$instances[$type][$entity] = new self();
         }
+        return self::$instances[$type][$entity];
+    }
 
-        static $instances = [];
-
-        if (!isset($instances[$type][$entity])) {
-            $instances[$type][$entity] = new self();
+    /**
+     * Reset singleton instances to force reload of rules.
+     */
+    public static function resetInstances(): void
+    {
+        foreach (self::$instances as $type_instances) {
+            foreach ($type_instances as $entity_instance) {
+                $entity_instance->load = 0;
+                $entity_instance->list = [];
+            }
         }
-        return $instances[$type][$entity];
     }
 }
```